core/src/test/java/hudson/FilePathTest.java

```diff
@@ -879,4 +881,79 @@ public void testCreateTempDir() throws IOException, InterruptedException  {
+
+        try (OutputStream os = tarFilePath.write()) {
+            new FilePath(folderToTar.toFile()).tar(os, "**/*");
+        }
+
+        Path untarredFolder = temp.getRoot().toPath().resolve("untarredFolder1");
+        tarFilePath.untar(new FilePath(untarredFolder.toFile()), FilePath.TarCompression.NONE);
+        assertTrue("build/lastSuccessfulBuild is not a symlink", Files.isSymbolicLink(untarredFolder.resolve("build/lastSuccessfulBuild")));
+    }
+
+    @Issue("JENKINS-52781")
+    @Test public void aSymlinkToADirectoryIsKeptByTarArchiveWithFileFilter() throws IOException, InterruptedException {
+        // create a simple hierarchy resembling a Jenkins build directory
+        Path folderToTar = temp.newFolder("tarme").toPath();
+        Path subdirectory = Files.createDirectory(folderToTar.resolve("build"));
+        Path targetDirectory = subdirectory.resolve("1");
+        Files.createDirectory(targetDirectory);
+        Files.createFile(targetDirectory.resolve("logs"));
+        Files.createSymbolicLink(subdirectory.resolve("lastSuccessfulBuild"), targetDirectory);
+        Path tarFile = temp.getRoot().toPath().resolve("archive.tar");
+        FilePath tarFilePath = new FilePath(tarFile.toFile());
+
+        try (OutputStream os = tarFilePath.write()) {
+            new FilePath(folderToTar.toFile()).tar(os, TrueFileFilter.TRUE);
+        }
+
+        Path untarredFolder = temp.getRoot().toPath().resolve("untarredFolder2");
+        tarFilePath.untar(new FilePath(untarredFolder.toFile()), FilePath.TarCompression.NONE);
+        assertTrue("tarme/build/lastSuccessfulBuild is not a symlink", Files.isSymbolicLink(untarredFolder.resolve("tarme/build/lastSuccessfulBuild")));
+    }
+
+    @Issue("JENKINS-52781")
+    @Test public void aSymlinkToAFileIsKeptByTarArchive() throws IOException, InterruptedException {
+        Path folderToTar = temp.newFolder().toPath();
+        Path subdirectory = Files.createDirectory(folderToTar.resolve("build"));
+        Path targetFile = subdirectory.resolve("1");
+        Files.createFile(targetFile);
+        Files.createSymbolicLink(subdirectory.resolve("lastSuccessfulBuild"), targetFile.resolve("1"));
+        Path tarFile = temp.getRoot().toPath().resolve("archive.tar");
+        FilePath tarFilePath = new FilePath(tarFile.toFile());
+
+        try (OutputStream os = tarFilePath.write()) {
+            new FilePath( folderToTar.toFile()).tar(os, "**/*");
+        }
+
+        Path untarredFolder = temp.getRoot().toPath().resolve("untarredFolder3");
+        tarFilePath.untar(new FilePath(untarredFolder.toFile()), FilePath.TarCompression.NONE);
+        assertTrue("build/lastSuccessfulBuild is not a symlink", Files.isSymbolicLink(untarredFolder.resolve("build/lastSuccessfulBuild")));
+
+    }
+
+    @Issue("JENKINS-52781")
+    @Test public void tryingToCreateADirectoryOnAnExistingDirectorySymlinkShouldNotFailAndAlsoCreateTheTargetDirectory() throws IOException {
```

Comment:
I guess my question about `mkdirs` boils down to whether this test case is the expected behavior or not. I would be interested to hear what other reviewers think.

Reply:
@dwnusbaum do you have ideas on who to ping to get this moving?